### PR TITLE
feat: add report level stub strategy

### LINKED ISSUE #305: Cannot see and update report stub strategy
Like in the old GUI:

image

### LINKED ISSUE #590: I cannot see and edit the stub strategy of a report
In Debug tab:

image

image

And when opening new tab:

image

image

### PR DESCRIPTION
**Pull request by MatthijsSmets**
This [PR](https://github.com/wearefrank/ladybug/pull/287) on the backend has to be merged first.
Closes #305 
Closes #590

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -158,4 +148,10 @@ export class AppComponent implements OnInit, OnDestroy, AfterViewInit {
     event.preventDefault();
     this.closeTab(tab);
   }
+
+  getStubStrategies(): void {
+    this.httpService
+      .getStubStrategies()
+      .subscribe((response: string[]) => (StubStrategyUtil.reportStubStrategies = response));
+  }
```
Comment (MarkvdVorst) [inline src/app/app.component.ts:156]: If new instances of the API call methods are used, please add the new error handling that is being used in #609

File: src/app/report/edit-display/edit-display.component.html [hunk 2]
```diff
@@ -69,15 +69,26 @@
...
-        <div class="w-fit mx-1 my-0">
-          <select class="form-control" [(ngModel)]="stubStrategy" (ngModelChange)="openStubDifferenceModal($event)">
-            <option [ngValue]="-1" [selected]="selectedNode.stub === -1">Use report level stub strategy</option>
-            <option [ngValue]="0" [selected]="selectedNode.stub === 0">Always stub this checkpoint</option>
-            <option [ngValue]="1" [selected]="selectedNode.stub === 1">Never stub this checkpoint</option>
-          </select>
-        </div>
+
+        @if (selectedNode) {
+          @if (ReportUtil.isReport(selectedNode)) {
```
Comment (MarkvdVorst) [inline src/app/report/edit-display/edit-display.component.html:74]: ReportUtil.isReport should already return false if selectedNode is false, so this initial @if can be removed.

**Review by MarkvdVorst — CHANGES_REQUESTED** [review]
Two small things to change

File: src/app/report/edit-display/edit-display.component.html (hunk 2 above)
Comment (MatthijsSmets) [reply src/app/report/edit-display/edit-display.component.html:74]: You're right, changed it.

File: src/app/app.component.ts (hunk 1 above)
Comment (MatthijsSmets) [reply src/app/app.component.ts:156]: Added it

File: src/app/shared/enums/stub-strategy.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export let StubStrategyUtil = {
```
Comment (ricklue) [inline src/app/shared/enums/stub-strategy.ts:1]: This is an enum, but the name says this is an util

Comment (MatthijsSmets) [reply src/app/shared/enums/stub-strategy.ts]: You're right, changed the name

**Review by ricklue — CHANGES_REQUESTED** [review]

File: src/app/report/edit-display/edit-display.component.ts [hunk 4]
```diff
@@ -68,7 +69,8 @@ import { DebugTabService } from '../../debug/debug-tab.service';
 export class EditDisplayComponent {
   protected readonly ReportUtil = ReportUtil;
   protected readonly Number: NumberConstructor = Number;
-  @Input() id: string = '';
+  protected readonly StubStrategyUtil = StubStrategy;
```
[the remark this answers is not in the document]
Comment (MatthijsSmets) [reply src/app/report/edit-display/edit-display.component.ts]: Fixed

File: src/app/report/edit-display/edit-display.component.html [hunk 5]
```diff
@@ -69,15 +69,25 @@
...
-            <option [ngValue]="0" [selected]="selectedNode.stub === 0">Always stub this checkpoint</option>
-            <option [ngValue]="1" [selected]="selectedNode.stub === 1">Never stub this checkpoint</option>
-          </select>
-        </div>
+
+
+        @if (ReportUtil.isReport(selectedNode)) {
+          <div class="w-fit mx-1 my-0">
+            <select class="form-control" [(ngModel)]="stubStrategy" (ngModelChange)="updateReportStubStrategy($event)">
+              @for (strategy of StubStrategyUtil.report; track strategy) {
```
[the remark this answers is not in the document]
Comment (MatthijsSmets) [reply src/app/report/edit-display/edit-display.component.html]: Fixed

**Review by MarkvdVorst — APPROVED** [review]

**Pull request merged**